src/appStartup.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AppStartupContext } from '@azure/functions-core';
+import { pathExists } from 'fs-extra';
+import { AzureFunctionsRpcMessages as rpc } from '../azure-functions-language-worker-protobuf/src/rpc';
+import { loadScriptFile } from './loadScriptFile';
+import { ensureErrorType } from './utils/ensureErrorType';
+import { WorkerChannel } from './WorkerChannel';
+import path = require('path');
+import LogLevel = rpc.RpcLog.Level;
+import LogCategory = rpc.RpcLog.RpcLogCategory;
+
+/**
+ * App startup can happen in two places, depending on if the worker was specialized or not
+ * 1. The worker can start in "normal" mode, meaning `workerInitRequest` will reference the user's app
+ * 2. The worker can start in "placeholder" mode, meaning `workerInitRequest` will reference a dummy app to "warm up" the worker and `functionEnvironmentReloadRequest` will be sent with the user's actual app.
+ *    This process is called worker specialization and it helps with cold start times.
+ *    The dummy app should never have actual startup code, so it should be safe to call `appStartup` twice in this case
+ *    Worker specialization happens only once, so we don't need to worry about cleaning up resources from previous `functionEnvironmentReloadRequest`s.
+ */
+export async function appStartup(functionAppDirectory: string, channel: WorkerChannel): Promise<void> {
```

Comment:
nit: It's common naming convention that a function in TypeScript should always start with a verb. Meaning this should be `startupApp`, `startApp`, or `initApp`.

Reply:
Renamed to `startApp`